src/textual/widgets/_data_table.py

```diff
@@ -950,7 +955,63 @@ def watch_cursor_coordinate(
             elif self.cursor_type == "column":
                 self.refresh_column(old_coordinate.column)
                 self._highlight_column(new_coordinate.column)
-            self._scroll_cursor_into_view()
+            # If the coordinate was changed via `move_cursor`, give priority to its
+            # scrolling because it may be animated.
+            self.call_next(self._scroll_cursor_into_view)
+
+    def move_cursor(
+        self,
+        coordinate: Coordinate | None = None,
+        *,
+        row: int | None = None,
+        column: int | None = None,
+        animate: bool = False,
+    ) -> None:
+        """Move the cursor to the given position.
+
+        The new cursor position can be specified as a coordinate or you can specify
+        only the new row/column for the cursor.
+        Specifying a coordinate is mutually exclusive with specifying either a new
+        row or new column.
+
+        Example:
+            ```py
+            datatable = app.query_one(DataTable)
+            datatable.move_cursor(Coordinate(5, 7))
+            # datatable.cursor_coordinate == Coordinate(5, 7)
+            datatable.move_cursor(row=3)
+            # datatable.cursor_coordinate == Coordinate(3, 7)
+            datatable.move_cursor(row=4, column=6)
+            # datatable.cursor_coordinate == Coordinate(4, 6)
+            datatable.move_cursor(Coordinate(0, 0), row=3)  # RuntimeError
+            ```
+
+        Args:
+            coordinate: The new coordinate to move the cursor to.
+            row: The new row to move the cursor to.
+            column: The new column to move the cursor to.
+            animate: Whether to animate the change of coordinates.
+
+        Raises:
+            RuntimeError: If the parameter `coordinate` is specified together with any
+                of `row` or `column`, or if no new position is specified.
+        """
+        if coordinate is not None and (row is not None or column is not None):
+            raise RuntimeError("Can't specify `coordinate` and `row`/`column`.")
```

Comment:
I gather this is trying to resolve ambiguous combination of parameters. But it would be better to not have to catch these at runtime.